manager/allocator/allocator_test.go

```diff
@@ -1214,37 +1214,75 @@ func TestNodeAllocator(t *testing.T) {
 		}
 		assert.NoError(t, store.CreateNetwork(tx, n1))
 
+		// this network will never be used for any task
+		nUnused := &api.Network{
+			ID: "overlayIDUnused",
+			Spec: api.NetworkSpec{
+				Annotations: api.Annotations{
+					Name: "overlayIDUnused",
+				},
+			},
+		}
+		assert.NoError(t, store.CreateNetwork(tx, nUnused))
+
 		assert.NoError(t, store.CreateNode(tx, node1))
+
 		return nil
 	}))
 
 	nodeWatch, cancel := state.Watch(s.WatchQueue(), api.EventUpdateNode{}, api.EventDeleteNode{})
 	defer cancel()
 	netWatch, cancel := state.Watch(s.WatchQueue(), api.EventUpdateNetwork{}, api.EventDeleteNetwork{})
 	defer cancel()
+	taskWatch, cancel := state.Watch(s.WatchQueue(), api.EventUpdateTask{})
+	defer cancel()
 
 	// Start allocator
 	go func() {
 		assert.NoError(t, a.Run(context.Background()))
 	}()
 	defer a.Stop()
 
+	assert.NoError(t, s.Update(func(tx store.Tx) error {
+		// create a task assigned to this node that has a network attachment on
+		// n1
+		t1 := &api.Task{
+			ID:           "task1",
+			NodeID:       node1.ID,
+			DesiredState: api.TaskStateRunning,
+			Spec: api.TaskSpec{
+				Networks: []*api.NetworkAttachmentConfig{
+					{
+						Target: "overlayID1",
+					},
+				},
+			},
+		}
+
+		return store.CreateTask(tx, t1)
+	}))
+
+	// validate that the task is created
+	watchTask(t, s, taskWatch, false, isValidTask)
+
 	// Validate node has 2 LB IP address (1 for each network).
 	watchNetwork(t, netWatch, false, isValidNetwork)                                      // ingress
 	watchNetwork(t, netWatch, false, isValidNetwork)                                      // overlayID1
+	watchNetwork(t, netWatch, false, isValidNetwork)                                      // overlayID1
```

Comment:
Is this last one validating that the `overlayIDunused` gets deleted from the network when task reconciliation occurs?   If so, maybe update the comment.  if not, then (pardon my confusion) what is the other network event here?

Reply:
yeah sorry, i copypasted that line.